.github/workflows/c-cpp.yml

```diff
@@ -20,19 +20,20 @@ jobs:
         run: ./build.sh --release
 
       - name: check-dynamatic
-        continue-on-error: true
         run: ninja -C build check-dynamatic
 
       - name: check-dynamatic-experimental
-        continue-on-error: true
+        if: success() || failure()
         run: ninja -C build check-dynamatic-experimental
 
       - name: integration-test
+        if: success() || failure()
         run: | 
           cd tools/integration
           python3 run_integration.py --ignore ignored_tests.txt
 
       - uses: actions/upload-artifact@v4
+        if: success() || failure()
```

Comment:
This also solves the problem that the artifact is not made if CI fails, as I mentioned places like [here](https://github.com/EPFL-LAP/dynamatic/pull/394#issuecomment-2827170596)